crates/volta-core/src/tool/package_global/uninstall.rs

```diff
@@ -0,0 +1,92 @@
+use super::metadata::{BinConfig, PackageConfig};
+use super::new_package_image_dir;
+use crate::error::{Context, ErrorKind, Fallible};
+use crate::fs::{dir_entry_match, remove_dir_if_exists, remove_file_if_exists};
+use crate::layout::volta_home;
+use crate::shim;
+use crate::style::success_prefix;
+use crate::sync::VoltaLock;
+use log::{info, warn};
+
+/// Uninstalls the specified package.
+///
+/// This removes:
+///
+/// - The JSON configuration files for both the package and its bins
+/// - The shims for the package bins
+/// - The package directory itself
+pub fn uninstall(name: &str) -> Fallible<()> {
+    let home = volta_home()?;
+    // Acquire a lock on the Volta directory, if possible, to prevent concurrent changes
+    let _lock = VoltaLock::acquire();
+
+    // If the package config file exists, use that to remove any installed bins and shims
+    let package_config_file = home.default_package_config_file(name);
+    let package_found = if package_config_file.exists() {
+        let package_config = PackageConfig::from_file(&package_config_file)?;
```

Comment:
Agreed! I generally prefer the approach of “do it and handle the ‘not found’ error” for FS operations. Especially since it’s entirely possible for the file to be deleted in between the `exists` check and the use of the file. The directory locking mitigates this slightly, but not entirely.

In this case, this block of code is migrated from the non-feature-flatted implementation of `uninstall` and I didn’t see a simple change to implement it that way. But I agree we should fix that (and probably root out any other spots where we check-then-open). I’ll make a follow-up issue

Reply:
Created https://github.com/volta-cli/volta/issues/823 to cover clean-up of _all_ of our FS operations to make sure we don't do the `exists() -> open()` flow.

Reply:
Ah, I’d missed that it was pulled from the existing implementation, and your reasoning is identical to mine. Thanks for creating the follow-up issue!